### PR TITLE
Support for cloud_firestore 4.4.0

### PR DESCRIPTION
**Pull request by sydneyagcaoili**
### What's in this PR

**Problem**
- With the latest cloud_firestore [changes](https://github.com/firebase/flutterfire/commit/9d91d513fad326f9c928d7d96d03e2c031875903), when using this version, signature related errors appeared.
- With [this](https://github.com/firebase/flutterfire/commit/538090fc49078b8d6c484d8db9049f06d05157dd) change, we need to support FieldPath on docRef.update

**Fix**
- convert relevant signatures to match cloud_firestore 's
- Add support for FieldPath params to docRef.update
- Fix test related issues
- Add test for FieldPath support on docRef.update

### CODE REVIEW

**Comment by sydneyagcaoili** [comment]
https://github.com/atn832/fake_cloud_firestore/issues/265 relevant issue

**Comment by atn832** [comment]
This looks very good. And thanks for the quick fix.

File: lib/src/mock_document_reference.dart [hunk 1]
```diff
@@ -103,7 +103,7 @@ class MockDocumentReference<T extends Object?> implements DocumentReference<T> {
         message: 'Some requested document was not found.',
       ));
     }
-    return _setRawData(data);
+    return _setRawData(Map.from(data));
```
Comment (atn832) [inline lib/src/mock_document_reference.dart:106]: What is this for?

**Comment by atn832** [comment]
Looks like all unit tests still pass. Awesome!

One more thing you could do in this PR is modify the affected pre-existing unit tests by passing iterables. In the Firestore PR you mentioned, they used this trick in their tests:
```dart
// To force the list to be an iterable
whereIn: ['Ready to Ship', 'Ordered'].map((e) => e),
```

We could make a utility function and use it here and there:
```dart
Iterable<R> toIterable<R>(List<R> list) => list.map((e) => e);
```

The tests in question should be in:
- https://github.com/atn832/fake_cloud_firestore/blob/master/test/mock_query_test.dart for the query change
- https://github.com/atn832/fake_cloud_firestore/blob/master/test/fake_cloud_firestore_test.dart for the docRef.update change.

File: lib/src/mock_document_reference.dart (hunk 1 above)
Comment (sydneyagcaoili) [reply lib/src/mock_document_reference.dart:106]: Due to new support to FieldPath on **update**, the Map<String, dynamic> is not gonna cut it.. Seems this needs more work.

**Comment by sydneyagcaoili** [comment]
@atn832 Agreed on supporting iterables.

Seems there needs to be more work on supporting FieldPath on update.  Not confident I'll be able to work on it quickly due to main commitments. But can do in my free time when no one else does.

**Pull request merged**